### PR TITLE
Fix syntect scip off-by-one

### LINKED ISSUE #57043: YAML Hyperlinks Omit The Last Number
- **Sourcegraph version:** v5.1.9
- **Platform information:** Web App (on all platforms)

#### Steps to reproduce:

1. Navigate to a YAML file with a hyperlink within it that ends in a digit.
 

#### Expected behavior:

The hyperlink is fully highlighted and clickable.

#### Actual behavior:

The hyperlink is not fully highlighted nor clickable.

image
Screen Shot 2023-09-26 at 10 52 52 AM

If you would like immediate help on this, please email support@sourcegraph.com (you can still create the issue, but there are no [SLAs](https://about.sourcegraph.com/support/) on issues like there are for support requests).

/cc @sourcegraph/search-platform

### PR DESCRIPTION
**Pull request by SuperAuguste**
Closes #57043 I believe

## Test plan

Snapshot tests passing locally.

### CODE REVIEW

**Review by cesrjimenez — APPROVED** [review]
🔥

**Review by jasonhawkharris — COMMENTED** [review]
Nice!

File: docker-images/syntax-highlighter/crates/sg-syntax/src/sg_sciptect.rs [hunk 1]
```diff
@@ -370,7 +370,10 @@ impl<'a> DocumentGenerator<'a> {
                     .find(|(_, (offset, _))| *offset == byte_offset)
                 {
                     Some(char) => char.0,
-                    None => line_contents.chars().count() - 1,
+                    None => {
+                        line_contents.chars().count()
+                            - (if line_contents.ends_with("\n") { 1 } else { 0 })
```
Comment (fkling) [inline docker-images/syntax-highlighter/crates/sg-syntax/src/sg_sciptect.rs:375]: This won't properly account for Windows-style line breaks though (`\r\n`), right? Though I don't know if code environments simply standardize on `\n`, even on Windows.

I guess it's not a big deal to include the `\r` for syntax highlighting purposes, but I still wonder if a "safer" solution would be to trim all whitespace/control characters from the end of the line.

Comment (SuperAuguste) [reply docker-images/syntax-highlighter/crates/sg-syntax/src/sg_sciptect.rs:375]: Ah good point! The old code didn't think of this either so I assumed it wasn't important but now I'm curious... 🤔

Comment (peterguy) [reply docker-images/syntax-highlighter/crates/sg-syntax/src/sg_sciptect.rs:375]: 👴 In my day, EOL was `\n`, `\r\n` or `\r`. 😄  pre-OSX Macintosh used only carriage return for EOL.

**Pull request merged**